framework/web/Request.php

```diff
@@ -163,7 +163,54 @@ class Request extends \yii\base\Request
      * @see getBodyParams()
      */
     public $parsers = [];
+    /**
+     * @var array The configuration for trusted security related headers.
+     * An array key is a regular expression for matching a hostname.
+     * A value is a list of headers to trust.
+     *
+     * For example, to trust all headers from domains ending in `.trusted.com` use the following:
+     *
+     * ```php
+     * [
+     *     '/^.*\.trusted\.com$/',
+     * ```
+     *
+     * To trust just the `x-forwarded-for` header from domains ending in `.partial.com` use:
+     *
+     * ```
+     *     // ...
+     *     '/^.*\.partial\.com$/' => ['X-Forwarded-For']
+     * ]
+     * ```
 
+     * Default is to trust all headers from all hosts.
```

Comment:
On second inspection, @cebe I think the comment is out of date. The default configuration will remove secure headers and allow all other headers from all hosts.